source/common/router/config_impl.h

```diff
@@ -211,6 +211,7 @@ class RouteEntryImplBase : public RouteEntry, public Matchable, public RedirectE
   const RateLimitPolicyImpl rate_limit_policy_;
   const ShadowPolicyImpl shadow_policy_;
   const Upstream::ResourcePriority priority_;
+  std::vector<Json::Object> config_headers_;
```

Comment:
No JSON parsing in hot path. Unpack at config load time to make sure everything is correct, and store in struct.